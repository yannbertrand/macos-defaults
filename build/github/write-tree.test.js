const fs = require('fs')
jest.mock('fs')

const writeTree = require('./write-tree')

const destinationPath = './dist'
const copiedFiles = ['.gitmoji-changelogrc', 'license']

describe('write-tree', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('no categories', () => {
    beforeEach(() => callWriteTree({}))

    it('should copy some static files', () => {
      copiedFiles.forEach(file => {
        const fileContent = readFile(`${destinationPath}/${file}`)
        expect(fileContent).toEqual(`copied:./templates/${file}`)
      })
    })

    it('should write an empty readme.md file using the home template', () => {
      const rootReadmeContent = readFile(`${destinationPath}/readme.md`)
      expect(rootReadmeContent).toMatchSnapshot()
    })
  })

  describe('one category, no page', () => {
    describe('with image', () => {
      beforeEach(() =>
        callWriteTree({
          categories: [
            {
              folder: 'category',
              name: 'Category',
              description: 'Category description.',
              image: {
                filename: 'category.png',
                width: 740,
                height: 80
              }
            }
          ]
        })
      )

      it('should copy some static files', () => {
        copiedFiles.forEach(file => {
          const fileContent = readFile(`${destinationPath}/${file}`)
          expect(fileContent).toEqual(`copied:./templates/${file}`)
        })
      })

      it('should write a readme.md file using the home template', () => {
        const rootReadmeContent = readFile(`${destinationPath}/readme.md`)
        expect(rootReadmeContent).toMatchSnapshot()
      })

      it('should write a category/readme.md file using the category template', () => {
        const categoryReadmeContent = readFile(
          `${destinationPath}/category/readme.md`
        )
        expect(categoryReadmeContent).toMatchSnapshot()
      })

      it('should copy the category.png example image to a category folder', () => {
        const originImagePath = readFile(
          `${destinationPath}/category/category.png`
        )
        expect(originImagePath).toMatchInlineSnapshot(
          `"copied:../../images/category/category.png"`
        )
      })
    })
  })

  describe('one category, one page', () => {
    describe('with text example', () => {
      describe('with special chars values', () => {
        beforeEach(() =>
          callWriteTree({
            categories: [
              {
                folder: 'category',
                name: 'Category',
                description: 'Category description.',
                keys: [
                  {
                    key: 'page',
                    domain: 'com.apple.category',
                    title: 'Page',
                    description: 'Page description.',
                    param: { type: 'string' },
                    examples: [
                      {
                        value: '~/Desktop',
                        default: true,
                        text: 'output when value is ~/Desktop'
                      },
                      {
                        value: '~/Pictures',
                        text: 'output when value is ~/Pictures'
                      }
                    ],
                    versions: ['Big Sur']
                  }
                ]
              }
            ]
          })
        )

        it('should copy some static files', () => {
          copiedFiles.forEach(file => {
            const fileContent = readFile(`${destinationPath}/${file}`)
            expect(fileContent).toEqual(`copied:./templates/${file}`)
          })
        })

        it('should write a readme.md file using the home template', () => {
          const rootReadmeContent = readFile(`${destinationPath}/readme.md`)
          expect(rootReadmeContent).toMatchSnapshot()
        })

        it('should write a category/readme.md file using the category template', () => {
          const categoryReadmeContent = readFile(
            `${destinationPath}/category/readme.md`
          )
          expect(categoryReadmeContent).toMatchSnapshot()
        })

        it('should write a category/page/readme.md file using the page template', () => {
          const pageReadmeContent = readFile(
            `${destinationPath}/category/page/readme.md`
          )
          expect(pageReadmeContent).toMatchSnapshot()
        })
      })

      describe('with requirements', () => {
        beforeEach(() =>
          callWriteTree({
            categories: [
              {
                folder: 'category',
                name: 'Category',
                description: 'Category description.',
                keys: [
                  {
                    key: 'page',
                    domain: 'com.apple.category',
                    title: 'Page',
                    description: 'Page description.',
                    param: { type: 'string' },
                    requirements: [
                      {
                        folder: 'another-category',
                        key: 'another-key',
                        name: 'com.apple.category2 another-key',
                        value: true
                      },
                      {
                        folder: 'a-third-category',
                        key: 'a-third-key',
                        name: 'com.apple.category2 a-third-key',
                        value: 0.5
                      },
                    ],
                    examples: [
                      {
                        value: true,
                        default: true,
                        text: 'output when value is true'
                      },
                      {
                        value: false,
                        text: 'output when value is false'
                      }
                    ],
                    versions: ['Big Sur'],
                    after: 'killall App'
                  }
                ]
              }
            ]
          })
        )

        it('should copy some static files', () => {
          copiedFiles.forEach(file => {
            const fileContent = readFile(`${destinationPath}/${file}`)
            expect(fileContent).toEqual(`copied:./templates/${file}`)
          })
        })

        it('should write a readme.md file using the home template', () => {
          const rootReadmeContent = readFile(`${destinationPath}/readme.md`)
          expect(rootReadmeContent).toMatchSnapshot()
        })

        it('should write a category/readme.md file using the category template', () => {
          const categoryReadmeContent = readFile(
            `${destinationPath}/category/readme.md`
          )
          expect(categoryReadmeContent).toMatchSnapshot()
        })

        it('should write a category/page/readme.md file using the page template', () => {
          const pageReadmeContent = readFile(
            `${destinationPath}/category/page/readme.md`
          )
          expect(pageReadmeContent).toMatchSnapshot()
        })
      })

      describe('with an after command', () => {
        beforeEach(() =>
          callWriteTree({
            categories: [
              {
                folder: 'category',
                name: 'Category',
                description: 'Category description.',
                keys: [
                  {
                    key: 'page',
                    domain: 'com.apple.category',
                    title: 'Page',
                    description: 'Page description.',
                    param: { type: 'string' },
                    examples: [
                      {
                        value: '~/Desktop',
                        default: true,
                        text: 'output when value is ~/Desktop'
                      },
                      {
                        value: '~/Pictures',
                        text: 'output when value is ~/Pictures'
                      }
                    ],
                    versions: ['Big Sur'],
                    after: 'killall App'
                  }
                ]
              }
            ]
          })
        )

        it('should copy some static files', () => {
          copiedFiles.forEach(file => {
            const fileContent = readFile(`${destinationPath}/${file}`)
            expect(fileContent).toEqual(`copied:./templates/${file}`)
          })
        })

        it('should write a readme.md file using the home template', () => {
          const rootReadmeContent = readFile(`${destinationPath}/readme.md`)
          expect(rootReadmeContent).toMatchSnapshot()
        })

        it('should write a category/readme.md file using the category template', () => {
          const categoryReadmeContent = readFile(
            `${destinationPath}/category/readme.md`
          )
          expect(categoryReadmeContent).toMatchSnapshot()
        })

        it('should write a category/page/readme.md file using the page template', () => {
          const pageReadmeContent = readFile(
            `${destinationPath}/category/page/readme.md`
          )
          expect(pageReadmeContent).toMatchSnapshot()
        })
      })
    })

    describe('with text and possible values example', () => {
      beforeEach(() =>
        callWriteTree({
          categories: [
            {
              folder: 'category',
              name: 'Category',
              description: 'Category description.',
              keys: [
                {
                  key: 'page',
                  domain: 'com.apple.category',
                  title: 'Page',
                  description: 'Page description.',
                  param: { type: 'string', values: ['start', 'middle', 'end'] },
                  examples: [
                    {
                      value: 'start',
                      text: 'output when value is start'
                    },
                    {
                      value: 'middle',
                      default: true,
                      text: 'output when value is middle'
                    },
                    {
                      value: 'end',
                      text: 'output when value is end'
                    }
                  ],
                  versions: ['Big Sur']
                }
              ]
            }
          ]
        })
      )

      it('should copy some static files', () => {
        copiedFiles.forEach(file => {
          const fileContent = readFile(`${destinationPath}/${file}`)
          expect(fileContent).toEqual(`copied:./templates/${file}`)
        })
      })

      it('should write a readme.md file using the home template', () => {
        const rootReadmeContent = readFile(`${destinationPath}/readme.md`)
        expect(rootReadmeContent).toMatchSnapshot()
      })

      it('should write a category/readme.md file using the category template', () => {
        const categoryReadmeContent = readFile(
          `${destinationPath}/category/readme.md`
        )
        expect(categoryReadmeContent).toMatchSnapshot()
      })

      it('should write a category/page.md file using the page template', () => {
        const pageReadmeContent = readFile(
          `${destinationPath}/category/page/readme.md`
        )
        expect(pageReadmeContent).toMatchSnapshot()
      })
    })

    describe('with image example', () => {
      beforeEach(() =>
        callWriteTree({
          categories: [
            {
              folder: 'category',
              name: 'Category',
              description: 'Category description.',
              keys: [
                {
                  key: 'page',
                  domain: 'com.apple.category',
                  title: 'Page',
                  description: 'Page description.',
                  param: { type: 'bool' },
                  examples: [
                    {
                      value: true,
                      default: true,
                      image: {
                        filename: 'true.png',
                        width: 600,
                        height: 400
                      }
                    },
                    {
                      value: false,
                      image: {
                        filename: 'false.png',
                        width: 400,
                        height: 200
                      }
                    }
                  ],
                  versions: ['Big Sur']
                }
              ]
            }
          ]
        })
      )

      it('should copy some static files', () => {
        copiedFiles.forEach(file => {
          const fileContent = readFile(`${destinationPath}/${file}`)
          expect(fileContent).toEqual(`copied:./templates/${file}`)
        })
      })

      it('should write a readme.md file using the home template', () => {
        const rootReadmeContent = readFile(`${destinationPath}/readme.md`)
        expect(rootReadmeContent).toMatchSnapshot()
      })

      it('should write a category/readme.md file using the category template', () => {
        const categoryReadmeContent = readFile(
          `${destinationPath}/category/readme.md`
        )
        expect(categoryReadmeContent).toMatchSnapshot()
      })

      it('should write a category/page/readme.md file using the page template', () => {
        const pageReadmeContent = readFile(
          `${destinationPath}/category/page/readme.md`
        )
        expect(pageReadmeContent).toMatchSnapshot()
      })

      it('should copy the true.png example image to a category/page folder', () => {
        const originImagePath = readFile(
          `${destinationPath}/category/page/true.png`
        )
        expect(originImagePath).toMatchInlineSnapshot(
          `"copied:../../images/category/page/true.png"`
        )
      })

      it('should copy the false.png example image to a category/page folder', () => {
        const originImagePath = readFile(
          `${destinationPath}/category/page/false.png`
        )
        expect(originImagePath).toMatchInlineSnapshot(
          `"copied:../../images/category/page/false.png"`
        )
      })
    })

    describe('with video example', () => {
      beforeEach(() =>
        callWriteTree({
          categories: [
            {
              folder: 'category',
              name: 'Category',
              description: 'Category description.',
              keys: [
                {
                  key: 'page',
                  domain: 'com.apple.category',
                  title: 'Page',
                  description: 'Page description.',
                  param: { type: 'float' },
                  examples: [
                    {
                      value: '0',
                      default: true,
                      video: {
                        filename: '0.mp4',
                        width: 750,
                        height: 400
                      }
                    },
                    {
                      value: '0.5',
                      video: {
                        filename: '0.5.mp4',
                        width: 720,
                        height: 390
                      }
                    }
                  ],
                  versions: ['Big Sur']
                }
              ]
            }
          ]
        })
      )

      it('should copy some static files', () => {
        copiedFiles.forEach(file => {
          const fileContent = readFile(`${destinationPath}/${file}`)
          expect(fileContent).toEqual(`copied:./templates/${file}`)
        })
      })

      it('should write a readme.md file using the home template', () => {
        const rootReadmeContent = readFile(`${destinationPath}/readme.md`)
        expect(rootReadmeContent).toMatchSnapshot()
      })

      it('should write a category/readme.md file using the category template', () => {
        const categoryReadmeContent = readFile(
          `${destinationPath}/category/readme.md`
        )
        expect(categoryReadmeContent).toMatchSnapshot()
      })

      it('should write a category/page/readme.md file using the page template', () => {
        const pageReadmeContent = readFile(
          `${destinationPath}/category/page/readme.md`
        )
        expect(pageReadmeContent).toMatchSnapshot()
      })

      it('should copy the 0.mp4 example video to a category/page folder', () => {
        const originVideoPath = readFile(
          `${destinationPath}/category/page/0.mp4`
        )
        expect(originVideoPath).toMatchInlineSnapshot(
          `"copied:../../images/category/page/0.mp4"`
        )
      })

      it('should copy the 0.5.mp4 example video to a category/page folder', () => {
        const originVideoPath = readFile(
          `${destinationPath}/category/page/0.5.mp4`
        )
        expect(originVideoPath).toMatchInlineSnapshot(
          `"copied:../../images/category/page/0.5.mp4"`
        )
      })
    })
  })

  describe('one category, two pages', () => {
    beforeEach(() =>
      callWriteTree({
        categories: [
          {
            folder: 'category',
            name: 'Category',
            description: 'Category description.',
            keys: [
              {
                key: 'page1',
                domain: 'com.apple.category',
                title: 'Page 1',
                description: 'Page 1 description.',
                param: { type: 'bool' },
                examples: [
                  {
                    value: true,
                    default: true,
                    text: 'output when value is true'
                  },
                  {
                    value: false,
                    text: 'output when value is false'
                  }
                ],
                versions: ['Big Sur']
              },
              {
                key: 'page2',
                domain: 'com.apple.category',
                title: 'Page 2',
                description: 'Page 2 description.',
                param: { type: 'bool' },
                examples: [
                  {
                    value: true,
                    text: 'output when value is true'
                  },
                  {
                    value: false,
                    default: true,
                    text: 'output when value is false'
                  }
                ],
                versions: ['Big Sur']
              }
            ]
          }
        ]
      })
    )

    it('should copy some static files', () => {
      copiedFiles.forEach(file => {
        const fileContent = readFile(`${destinationPath}/${file}`)
        expect(fileContent).toEqual(`copied:./templates/${file}`)
      })
    })

    it('should write a readme.md file using the home template', () => {
      const rootReadmeContent = readFile(`${destinationPath}/readme.md`)
      expect(rootReadmeContent).toMatchSnapshot()
    })

    it('should write a category/readme.md file using the category template', () => {
      const categoryReadmeContent = readFile(
        `${destinationPath}/category/readme.md`
      )
      expect(categoryReadmeContent).toMatchSnapshot()
    })

    it('should write a category/page1/readme.md file using the page template', () => {
      const pageReadmeContent = readFile(
        `${destinationPath}/category/page1/readme.md`
      )
      expect(pageReadmeContent).toMatchSnapshot()
    })

    it('should write a category/page2/readme.md file using the page template', () => {
      const pageReadmeContent = readFile(
        `${destinationPath}/category/page2/readme.md`
      )
      expect(pageReadmeContent).toMatchSnapshot()
    })
  })

  describe('two categories, one page in each', () => {
    beforeEach(() =>
      callWriteTree({
        categories: [
          {
            folder: 'category1',
            name: 'Category 1',
            description: 'Category 1 description.',
            keys: [
              {
                key: 'page',
                domain: 'com.apple.category1',
                title: 'Page',
                description: 'Page description.',
                param: { type: 'bool' },
                examples: [
                  {
                    value: true,
                    default: true,
                    text: 'output when value is true'
                  },
                  {
                    value: false,
                    text: 'output when value is false'
                  }
                ],
                versions: ['Big Sur']
              }
            ]
          },
          {
            folder: 'category2',
            name: 'Category 2',
            description: 'Category 2 description.',
            keys: [
              {
                key: 'page',
                domain: 'com.apple.category2',
                title: 'Page',
                description: 'Page description.',
                param: { type: 'bool' },
                examples: [
                  {
                    value: true,
                    text: 'output when value is true'
                  },
                  {
                    value: false,
                    default: true,
                    text: 'output when value is false'
                  }
                ],
                versions: ['Big Sur']
              }
            ]
          }
        ]
      })
    )

    it('should copy some static files', () => {
      copiedFiles.forEach(file => {
        const fileContent = readFile(`${destinationPath}/${file}`)
        expect(fileContent).toEqual(`copied:./templates/${file}`)
      })
    })

    it('should write a readme.md file using the home template', () => {
      const rootReadmeContent = readFile(`${destinationPath}/readme.md`)
      expect(rootReadmeContent).toMatchSnapshot()
    })

    it('should write a category1/readme.md file using the category template', () => {
      const categoryReadmeContent = readFile(
        `${destinationPath}/category1/readme.md`
      )
      expect(categoryReadmeContent).toMatchSnapshot()
    })

    it('should write a category2/readme.md file using the category template', () => {
      const categoryReadmeContent = readFile(
        `${destinationPath}/category2/readme.md`
      )
      expect(categoryReadmeContent).toMatchSnapshot()
    })

    it('should write a category1/page/readme.md file using the page template', () => {
      const pageReadmeContent = readFile(
        `${destinationPath}/category1/page/readme.md`
      )
      expect(pageReadmeContent).toMatchSnapshot()
    })

    it('should write a category2/page/readme.md file using the page template', () => {
      const pageReadmeContent = readFile(
        `${destinationPath}/category2/page/readme.md`
      )
      expect(pageReadmeContent).toMatchSnapshot()
    })
  })
})

const callWriteTree = sourceFile => writeTree(sourceFile, destinationPath)
const readFile = file => fs.readFakeFileSync(file, 'utf8')
