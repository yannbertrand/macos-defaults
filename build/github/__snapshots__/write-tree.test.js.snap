// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`write-tree no categories should write an empty readme.md file using the home template 1`] = `
"# macos-defaults
Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it from scratch. Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/github#readme) if you want to do the same.
"
`;

exports[`write-tree one category, no page with image should write a category/readme.md file using the category template 1`] = `
"# Category

![A screenshot of the Category](category.png)

> Category description.

"
`;

exports[`write-tree one category, no page with image should write a readme.md file using the home template 1`] = `
"# macos-defaults
Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## [Category](./category/readme.md)

## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it from scratch. Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/github#readme) if you want to do the same.
"
`;

exports[`write-tree one category, one page with image example should write a category/page/readme.md file using the page template 1`] = `
"# [Category](../readme.md)

## Page

> Page description.

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: bool

### Set to \`true\` (default value)
\`\`\`bash
defaults write com.apple.category page -bool true
\`\`\`
![Example output with value set to true](true.png)

### Set to \`false\`
\`\`\`bash
defaults write com.apple.category page -bool false
\`\`\`
![Example output with value set to false](false.png)

### Read current value
\`\`\`bash
defaults read com.apple.category page
\`\`\`

## Delete current value
\`\`\`bash
defaults delete com.apple.category page
\`\`\`
"
`;

exports[`write-tree one category, one page with image example should write a category/readme.md file using the category template 1`] = `
"# Category

> Category description.

- [Page](./page/readme.md)
"
`;

exports[`write-tree one category, one page with image example should write a readme.md file using the home template 1`] = `
"# macos-defaults
Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## [Category](./category/readme.md)
- [Page](./category/page/readme.md)

## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it from scratch. Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/github#readme) if you want to do the same.
"
`;

exports[`write-tree one category, one page with text and possible values example should write a category/page.md file using the page template 1`] = `
"# [Category](../readme.md)

## Page

> Page description.

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: string
  * start
  * middle
  * end

### Set to \`start\`
\`\`\`bash
defaults write com.apple.category page -string start
\`\`\`
- output when value is start

### Set to \`middle\` (default value)
\`\`\`bash
defaults write com.apple.category page -string middle
\`\`\`
- output when value is middle

### Set to \`end\`
\`\`\`bash
defaults write com.apple.category page -string end
\`\`\`
- output when value is end

### Read current value
\`\`\`bash
defaults read com.apple.category page
\`\`\`

## Delete current value
\`\`\`bash
defaults delete com.apple.category page
\`\`\`
"
`;

exports[`write-tree one category, one page with text and possible values example should write a category/readme.md file using the category template 1`] = `
"# Category

> Category description.

- [Page](./page/readme.md)
"
`;

exports[`write-tree one category, one page with text and possible values example should write a readme.md file using the home template 1`] = `
"# macos-defaults
Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## [Category](./category/readme.md)
- [Page](./category/page/readme.md)

## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it from scratch. Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/github#readme) if you want to do the same.
"
`;

exports[`write-tree one category, one page with text example with an after command should write a category/page/readme.md file using the page template 1`] = `
"# [Category](../readme.md)

## Page

> Page description.

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: string

### Set to \`~/Desktop\` (default value)
\`\`\`bash
defaults write com.apple.category page -string ~/Desktop && killall App
\`\`\`
- output when value is ~/Desktop

### Set to \`~/Pictures\`
\`\`\`bash
defaults write com.apple.category page -string ~/Pictures && killall App
\`\`\`
- output when value is ~/Pictures

### Read current value
\`\`\`bash
defaults read com.apple.category page
\`\`\`

## Delete current value
\`\`\`bash
defaults delete com.apple.category page && killall App
\`\`\`
"
`;

exports[`write-tree one category, one page with text example with an after command should write a category/readme.md file using the category template 1`] = `
"# Category

> Category description.

- [Page](./page/readme.md)
"
`;

exports[`write-tree one category, one page with text example with an after command should write a readme.md file using the home template 1`] = `
"# macos-defaults
Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## [Category](./category/readme.md)
- [Page](./category/page/readme.md)

## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it from scratch. Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/github#readme) if you want to do the same.
"
`;

exports[`write-tree one category, one page with text example with requirements should write a category/page/readme.md file using the page template 1`] = `
"# [Category](../readme.md)

## Page

> Page description.

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: string

### Requirements
- [\`com.apple.category2 another-key\`](../../another-category/another-key/readme.md#set-to-true) must be set to \`true\`
- [\`com.apple.category2 a-third-key\`](../../a-third-category/a-third-key/readme.md#set-to-0.5) must be set to \`0.5\`

### Set to \`true\` (default value)
\`\`\`bash
defaults write com.apple.category page -string true && killall App
\`\`\`
- output when value is true

### Set to \`false\`
\`\`\`bash
defaults write com.apple.category page -string false && killall App
\`\`\`
- output when value is false

### Read current value
\`\`\`bash
defaults read com.apple.category page
\`\`\`

## Delete current value
\`\`\`bash
defaults delete com.apple.category page && killall App
\`\`\`
"
`;

exports[`write-tree one category, one page with text example with requirements should write a category/readme.md file using the category template 1`] = `
"# Category

> Category description.

- [Page](./page/readme.md)
"
`;

exports[`write-tree one category, one page with text example with requirements should write a readme.md file using the home template 1`] = `
"# macos-defaults
Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## [Category](./category/readme.md)
- [Page](./category/page/readme.md)

## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it from scratch. Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/github#readme) if you want to do the same.
"
`;

exports[`write-tree one category, one page with text example with special chars values should write a category/page/readme.md file using the page template 1`] = `
"# [Category](../readme.md)

## Page

> Page description.

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: string

### Set to \`~/Desktop\` (default value)
\`\`\`bash
defaults write com.apple.category page -string ~/Desktop
\`\`\`
- output when value is ~/Desktop

### Set to \`~/Pictures\`
\`\`\`bash
defaults write com.apple.category page -string ~/Pictures
\`\`\`
- output when value is ~/Pictures

### Read current value
\`\`\`bash
defaults read com.apple.category page
\`\`\`

## Delete current value
\`\`\`bash
defaults delete com.apple.category page
\`\`\`
"
`;

exports[`write-tree one category, one page with text example with special chars values should write a category/readme.md file using the category template 1`] = `
"# Category

> Category description.

- [Page](./page/readme.md)
"
`;

exports[`write-tree one category, one page with text example with special chars values should write a readme.md file using the home template 1`] = `
"# macos-defaults
Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## [Category](./category/readme.md)
- [Page](./category/page/readme.md)

## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it from scratch. Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/github#readme) if you want to do the same.
"
`;

exports[`write-tree one category, one page with video example should write a category/page/readme.md file using the page template 1`] = `
"# [Category](../readme.md)

## Page

> Page description.

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: float

### Set to \`0\` (default value)
\`\`\`bash
defaults write com.apple.category page -float 0
\`\`\`
[Link to a demo video](0.mp4)

### Set to \`0.5\`
\`\`\`bash
defaults write com.apple.category page -float 0.5
\`\`\`
[Link to a demo video](0.5.mp4)

### Read current value
\`\`\`bash
defaults read com.apple.category page
\`\`\`

## Delete current value
\`\`\`bash
defaults delete com.apple.category page
\`\`\`
"
`;

exports[`write-tree one category, one page with video example should write a category/readme.md file using the category template 1`] = `
"# Category

> Category description.

- [Page](./page/readme.md)
"
`;

exports[`write-tree one category, one page with video example should write a readme.md file using the home template 1`] = `
"# macos-defaults
Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## [Category](./category/readme.md)
- [Page](./category/page/readme.md)

## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it from scratch. Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/github#readme) if you want to do the same.
"
`;

exports[`write-tree one category, two pages should write a category/page1/readme.md file using the page template 1`] = `
"# [Category](../readme.md)

## Page 1

> Page 1 description.

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: bool

### Set to \`true\` (default value)
\`\`\`bash
defaults write com.apple.category page1 -bool true
\`\`\`
- output when value is true

### Set to \`false\`
\`\`\`bash
defaults write com.apple.category page1 -bool false
\`\`\`
- output when value is false

### Read current value
\`\`\`bash
defaults read com.apple.category page1
\`\`\`

## Delete current value
\`\`\`bash
defaults delete com.apple.category page1
\`\`\`
"
`;

exports[`write-tree one category, two pages should write a category/page2/readme.md file using the page template 1`] = `
"# [Category](../readme.md)

## Page 2

> Page 2 description.

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: bool

### Set to \`true\`
\`\`\`bash
defaults write com.apple.category page2 -bool true
\`\`\`
- output when value is true

### Set to \`false\` (default value)
\`\`\`bash
defaults write com.apple.category page2 -bool false
\`\`\`
- output when value is false

### Read current value
\`\`\`bash
defaults read com.apple.category page2
\`\`\`

## Delete current value
\`\`\`bash
defaults delete com.apple.category page2
\`\`\`
"
`;

exports[`write-tree one category, two pages should write a category/readme.md file using the category template 1`] = `
"# Category

> Category description.

- [Page 1](./page1/readme.md)
- [Page 2](./page2/readme.md)
"
`;

exports[`write-tree one category, two pages should write a readme.md file using the home template 1`] = `
"# macos-defaults
Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## [Category](./category/readme.md)
- [Page 1](./category/page1/readme.md)
- [Page 2](./category/page2/readme.md)

## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it from scratch. Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/github#readme) if you want to do the same.
"
`;

exports[`write-tree two categories, one page in each should write a category1/page/readme.md file using the page template 1`] = `
"# [Category 1](../readme.md)

## Page

> Page description.

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: bool

### Set to \`true\` (default value)
\`\`\`bash
defaults write com.apple.category1 page -bool true
\`\`\`
- output when value is true

### Set to \`false\`
\`\`\`bash
defaults write com.apple.category1 page -bool false
\`\`\`
- output when value is false

### Read current value
\`\`\`bash
defaults read com.apple.category1 page
\`\`\`

## Delete current value
\`\`\`bash
defaults delete com.apple.category1 page
\`\`\`
"
`;

exports[`write-tree two categories, one page in each should write a category1/readme.md file using the category template 1`] = `
"# Category 1

> Category 1 description.

- [Page](./page/readme.md)
"
`;

exports[`write-tree two categories, one page in each should write a category2/page/readme.md file using the page template 1`] = `
"# [Category 2](../readme.md)

## Page

> Page description.

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: bool

### Set to \`true\`
\`\`\`bash
defaults write com.apple.category2 page -bool true
\`\`\`
- output when value is true

### Set to \`false\` (default value)
\`\`\`bash
defaults write com.apple.category2 page -bool false
\`\`\`
- output when value is false

### Read current value
\`\`\`bash
defaults read com.apple.category2 page
\`\`\`

## Delete current value
\`\`\`bash
defaults delete com.apple.category2 page
\`\`\`
"
`;

exports[`write-tree two categories, one page in each should write a category2/readme.md file using the category template 1`] = `
"# Category 2

> Category 2 description.

- [Page](./page/readme.md)
"
`;

exports[`write-tree two categories, one page in each should write a readme.md file using the home template 1`] = `
"# macos-defaults
Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## [Category 1](./category1/readme.md)
- [Page](./category1/page/readme.md)

## [Category 2](./category2/readme.md)
- [Page](./category2/page/readme.md)

## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it from scratch. Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/github#readme) if you want to do the same.
"
`;
